src/posix/platform/settings.hpp

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
why do we have so much change in settings.hpp and settings.cpp, when the PR title seems to indicate we are only changing the way critical keys are initialized?

Reply:
These changes are from https://github.com/openthread/openthread/pull/7495. It's included here because this PR depends on it. Please review only the last commit.

Reply:
After some changes, this PR doesn't depend on that any longer. I removed that commit.